Automattic/vip-go-mu-plugins
# Streamwrapper: Upload file to fileservice on fopen

## Description

Upload a new file to the fileservice on `fopen()` as that is the expected behaviour for the `fopen()` function. This was causing problems with some plugins where they were checking the file with `is_file()` or `is_writable()` after `fopen()` was called.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Check out PR.
1. Deploy changes on a test site
1. `wp shell`
1. `fopen( 'vip://wp-content/uploads/testfile', 'w' );`
1. ensure `is_file()` and `is_writable()` on above file returns true.
1. `fclose( $handle );`
1. Ensure file is uploaded to file service
1. ensure existing filesystem functionality still works
